### PR TITLE
Add serializer for params

### PR DESCRIPTION
**Pull request by elgatovital**
Changes proposed in this pull request:
- Fix crash due to marshaling problem when moving app to background

@gnosis/mobile-devs

### CODE REVIEW

File: app/src/main/java/io/gnosis/safe/utils/ParamSerializer.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    init {
+        decodedDataAdapter = moshi.adapter(DataDecodedDto::class.java)
+        decodedValuesAdapter = moshi.adapter(Types.newParameterizedType(List::class.java, ValueDecodedDto::class.java))
+    }
+
+    fun serializeDecodedData(decodedData: DataDecodedDto): String {
+        return decodedDataAdapter.toJson(decodedData)
+    }
+
+    fun unserializeDecodedData(decodedDataString: String): DataDecodedDto? {
```
Comment (biafra23) [inline app/src/main/java/io/gnosis/safe/utils/ParamSerializer.kt:24]: I would call it `deserializeDecodedData()` and `deserializeDecodedValues()` but that's not a blocker.

File: app/src/main/res/navigation/main_nav.xml [hunk 2]
```diff
@@ -176,7 +176,7 @@
 
         <argument
             android:name="decodedData"
-            app:argType="java.io.Serializable" />
```
Comment (biafra23) [inline app/src/main/res/navigation/main_nav.xml:179]: Does that mean we cannot use Serializable arguments in general? Or is it a problem in this specific case only, and if so, why?

File: data/src/main/java/io/gnosis/data/backend/dto/TransactionDto.kt [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
     val method: String,
     val parameters: List<ParamDto>?
-) : Serializable
+)
 
 data class ValueDecodedDto(
     val operation: Operation,
     val to: Solidity.Address,
-    val value: Long,
+    val value: BigInteger,
```
Comment (biafra23) [inline data/src/main/java/io/gnosis/data/backend/dto/TransactionDto.kt:14]: 👍

**Review by biafra23 — APPROVED** [review]
LGTM 

Why did Serialize not work? Is it a general problem? Why would Android offer that method if it doesn't survive the lifecycle? Would Parcelable work or fail for the same reason?

**Comment by elgatovital** [comment]
> LGTM
> 
> Why did Serialize not work? Is it a general problem? Why would Android offer that method if it doesn't survive the lifecycle? Would Parcelable work or fail for the same reason?

I haven't investigated it in detail. Probably some child elements were not serializable.

**Pull request merged**